department-of-veterans-affairs/caseflow
# Distribute down to 30 legacy non priority cases tied to a judge before proceeding with auto case distribution

Resolves #14543

### Description
Ensures we are distributing as many legacy hearing cases tied to a judge

### Acceptance Criteria
- [ ] New step to distribute legacy non priority hearing backlog cases tied to a judge before continuing with the rest of the algorithm
   - [ ] If a judge has over 30 legacy hearing cases in their backlog, distribute down to 30 while still respecting the provided case order limit
- [ ] Rest of the algorithm is undisturbed
   - [ ] If the judge has < 30 cases, the algorithm distributes the exact cases it would have without the implemented changes
   - [ ] If the judge has > 30 cases, after distributing down to 30, the algorithm distributes the reminder of the case order as it would have without the implemented changes, but with the new lower limit. (judge orders 25 cases, 10 backlog cases are distributed, algorithm continues, distributing the remaining 15 as if that was the original case order)

### Testing Plan
1. Log in as a judge
1. Go to their assign page
1. Request more cases
1. Check the distribution for stats
```ruby
Distribution.last.statistics
=> {"batch_size"=>12,
 "total_batch_size"=>57,
 "priority_count"=>0,
 "direct_review_due_count"=>0,
 "legacy_hearing_backlog_count"=>0,     # <= Woo!
 "legacy_proportion"=>0.17663136317048175,
 "direct_review_proportion"=>0.0,
 "evidence_submission_proportion"=>0.4748889648103861,
 "hearing_proportion"=>0.3484796720191322,
 "pacesetting_direct_review_proportion"=>5.7272727272727275,
 "interpolated_minimum_direct_review_proportion"=>0.0,
 "nonpriority_iterations"=>2}
```
